servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/PublisherFlatMapSingle.java

```diff
@@ -399,27 +364,28 @@ public void onSuccess(@Nullable R result) {
 
             @Override
             public void onError(Throwable t) {
+                Throwable currPendingError = pendingError;
                 if (source.maxDelayedErrors == 0) {
-                    onError0(t, true, true);
+                    if (currPendingError == null &&
+                            pendingErrorUpdater.compareAndSet(FlatMapSubscriber.this, null, t)) {
```

Comment:
Inside the above `if` branch (line 368) `currPendingError` is never used. Consider simplifying this one to
```
if (pendingErrorUpdater.compareAndSet(FlatMapSubscriber.this, null, t)) {
```